golang/pkg/rnr/job.go

```diff
@@ -12,11 +12,11 @@ import (
 type Job struct {
 	pbMutex sync.Mutex
 	job     pb.Job
-	root    TaskInterface
```

Comment:
My concern here is that although the former felt a bit too verbose, renaming to `Task` might lead to confusion with structs such as `NestedTask` or `CallbackTask`. No strong opinion though.